detector.go

```diff
@@ -76,7 +74,7 @@ type DetectConfig struct {
 	AppDir        string
 	PlatformDir   string
 	BuildpacksDir string
-	Logger        logging.Logger
+	Logger        Logger
```

Comment:
does using unscoped `Logger` for both the instance and the type create a risk of collision?

Reply:
Given that the instance is a field I'm not worried about collision. Defining the interface type in the package that consumes it, rather than the package that provides the implementation is idiomatic.